src/main/java/no/nav/pto/veilarbportefolje/arenaaktiviteter/GruppeAktivitetRepository.java

```diff
@@ -0,0 +1,170 @@
+                ).execute();
+    }
+
+    /**
+     * Setter aktivitet til inaktiv kun hvis aktiviteten er utgatt.
+     * Implementert til aa forhindre race condition mellom daglig jobb og kafka.
+     */
+    public int oppdaterUtgattAktivStatus(String moteplanId, String veiledningdeltakerId, AktorId aktorId, PersonId personId) {
+        String updateSql = String.format(
+                "UPDATE %s SET %s = 'N' WHERE %s = ? AND %s = ? AND %s < CURRENT_TIMESTAMP",
+                TABLE_NAME, AKTIV, MOTEPLAN_ID, VEILEDNINGDELTAKER_ID, MOTEPLAN_SLUTTDATO
+        );
+
+        int updated = db.update(updateSql, moteplanId, veiledningdeltakerId);
+        if (updated != 0) {
+            utledOgLagreGruppeaktiviteter(personId, aktorId);
+        }
+        return updated;
+    }
+
+    public Optional<Long> retrieveHendelse(GruppeAktivitetInnhold aktivitet) {
+        return Optional.ofNullable(
+                SqlUtils.select(db, TABLE_NAME, rs -> rs.getLong(HENDELSE_ID))
+                        .column(HENDELSE_ID)
+                        .where(
+                                WhereClause.equals(MOTEPLAN_ID, aktivitet.getMoteplanId())
+                                        .and(WhereClause.equals(VEILEDNINGDELTAKER_ID, aktivitet.getVeiledningdeltakerId()))
+                        )
+                        .execute()
+        );
+    }
+
+    public Optional<GruppeAktivitetSchedueldDTO> hentAktivtet(String moteplanId, String veiledningdeltakerId) {
+        return Optional.ofNullable(
+                SqlUtils.select(db, TABLE_NAME, this::mapTilDto)
+                        .column("*")
+                        .where(WhereClause.equals(MOTEPLAN_ID, moteplanId)
+                                .and(WhereClause.equals(VEILEDNINGDELTAKER_ID, veiledningdeltakerId))
+                        ).execute()
+        );
+    }
+
+    public List<GruppeAktivitetSchedueldDTO> hentUtgatteAktivteter() {
+        String sql = "SELECT * FROM " + TABLE_NAME
+                + " WHERE " + MOTEPLAN_SLUTTDATO + " < CURRENT_TIMESTAMP";
+        return db.queryForList(sql)
+                .stream()
+                .map(result -> (ResultSet) result)
+                .map(this::mapTilDto)
+                .collect(toList());
+    }
+
+    public void utledOgLagreGruppeaktiviteter(PersonId personId, AktorId aktorId) {
+        List<GruppeAktivitetSchedueldDTO> gruppeAktiviteter = hentAktiveAktivteter(aktorId);
+        Timestamp nesteStart = gruppeAktiviteter.stream()
+                .filter(GruppeAktivitetSchedueldDTO::isAktiv)
+                .map(GruppeAktivitetSchedueldDTO::getAktivitetperiodeFra)
+                .min(Comparator.naturalOrder())
+                .orElse(null);
+        Timestamp nesteUtlopsdato = gruppeAktiviteter.stream()
+                .filter(GruppeAktivitetSchedueldDTO::isAktiv)
+                .map(GruppeAktivitetSchedueldDTO::getAktivitetperiodeTil)
+                .max(Comparator.naturalOrder())
```

Comment:
just to be sure, this is next expiration date? I am just wondering why we are using `max`